addons/event/models/event_registration.py

```diff
@@ -252,6 +243,20 @@ def _update_mail_schedulers(self):
     # MAILING / GATEWAY
     # ------------------------------------------------------------
 
+    def _message_compute_subject(self, **kwargs):
+        record_names = dict()
+
+        to_compute_registrations = self.browse([thread_id for thread_id in self.ids if not record_names.get(thread_id)])
+        named_registrations = to_compute_registrations.filtered(lambda registration: registration.name)
+        unnamed_registrations = to_compute_registrations - named_registrations
+
+        record_names.update({registration.id: _("%s registration for %s", registration.event_id.name, registration.name)
```

Comment:
Could maybe be simplified in a for loop ? Like
```
for registration in self:
    if registration.name:
        subjects[registration.id] = _(
            "%(event_name)s registration for %(attendee_name)s",
            event_name=registration.event_id.name,
            attendee_name=registration.name,
        )
    else:
        subjects[registration.id] = _(
            "%(event_name)s registration",
            event_name=registration.event_id.name,
        )

Reply:
Added proposal to use named parameters, easier for people that translation the sentence to have more context :)